homeassistant/components/nuki/config_flow.py

```diff
@@ -0,0 +1,78 @@
+"""Config flow to configure the Nuki integration."""
+
+from pynuki import NukiBridge
+from requests.exceptions import RequestException
+import voluptuous as vol
+
+from homeassistant import config_entries, exceptions
+from homeassistant.const import CONF_HOST, CONF_PORT, CONF_TOKEN
+
+from .const import (  # pylint: disable=unused-import
+    DEFAULT_PORT,
+    DEFAULT_TIMEOUT,
+    DOMAIN,
+)
+
+USER_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        vol.Optional(CONF_PORT, default=DEFAULT_PORT): vol.Coerce(int),
+        vol.Required(CONF_TOKEN): str,
+    }
+)
+
+
+async def validate_input(hass, data):
+    """Validate the user input allows us to connect.
+
+    Data has the keys from DATA_SCHEMA with values provided by the user.
+    """
+
+    bridge = await hass.async_add_executor_job(
```

Comment:
We don't do IO in the init of NukiBridge, there is no need for an executor

Reply:
https://github.com/pschmitt/pynuki/blob/master/pynuki/bridge.py#L31-L45

Reply:
When I try to remove the `executor_job` I'm getting the following warning:
```
WARNING (MainThread) [homeassistant.util.async_] Detected I/O inside the event loop. This is causing stability issues. Please report issue for nuki doing I/O at homeassistant/components/nuki/config_flow.py, line 31: bridge = NukiBridge
```

Using the following code snippet:
```
bridge = NukiBridge(
        data[CONF_HOST],
        data[CONF_TOKEN],
        data[CONF_PORT],
        True,
        DEFAULT_TIMEOUT,
    )
```

Reply:
I've also needed to move the block inside the try-statement, since it tries to connect as soon as the token is set if I understand the code correctly: https://github.com/pschmitt/pynuki/blob/24966829b27e81db9ffb4a1395ab09456cf86929/pynuki/bridge.py#L75